godot-macros/src/method_registration/mod.rs

```diff
@@ -0,0 +1,137 @@
+/*
+* This Source Code Form is subject to the terms of the Mozilla Public
+* License, v. 2.0. If a copy of the MPL was not distributed with this
+* file, You can obtain one at https://mozilla.org/MPL/2.0/.
+*/
+pub mod register_method;
+pub mod virtual_method_callback;
+
+use proc_macro2::{Ident, TokenStream as TokenStream2};
+use quote::quote;
+use venial::{Function, TyExpr};
+
+pub use register_method::gdext_register_method;
+pub use virtual_method_callback::gdext_virtual_method_callback;
+
+#[derive(Copy, Clone, Eq, PartialEq)]
+enum ReceiverType {
+    Ref,
+    Mut,
+    Static,
+}
+
+struct SignatureInfo {
+    pub method_name: Ident,
+    pub receiver_type: ReceiverType,
+    pub param_idents: Vec<Ident>,
+    pub param_types: Vec<TyExpr>,
+    pub ret_type: TokenStream2,
+}
+
+fn wrap_with_unpacked_params(class_name: &Ident, signature_info: &SignatureInfo) -> TokenStream2 {
+    let method_name = &signature_info.method_name;
+    let params = &signature_info.param_idents;
+
+    let instance = match &signature_info.receiver_type {
```

Comment:
Please name this variable more expressively; it's not an instance but a declaration for a variable `instance`.

For example: `instance_decl`.